### ISSUE TITLE
Notes list scroll position jumps down while scrolling up

### ISSUE THREAD

**Issue by personalizedrefrigerator** [laurent22/joplin#9036]
## Environment

Joplin version: 2.13.2
Platform: Desktop
OS specifics: Ubuntu 23.04

## Steps to reproduce

1. Open a notebook with a large number of notes
2. Scroll down and select a note
3. Slowly scroll up

https://github.com/laurent22/joplin/assets/46334387/e34c4384-f281-4cd5-b02c-d5396846afd0

## Describe what you expected to happen

The note list should scroll up without jumping back to its original position.

**Comment by laurent22** [comment]
I'm not able to replicate this on macOS. I wonder if it's some interaction between the keyboard being used to move from one item to another, and also being used to move the scrollbar up and down. Does it do this if you move the mouse outside of the list?

**Comment by personalizedrefrigerator** [comment]
It works fine if I scroll with the keyboard &mdash; it only seems to happen if I use touchpad (**edit**: or touchscreen) scrolling and scroll slowly. (As such, in these cases, moving the mouse outside of the list and scrolling does not scroll the list.)

**Comment by personalizedrefrigerator** [comment]
If you're still unable to reproduce this, I can try to fix it.

**Comment by laurent22** [comment]
Oh I see, for me it definitely doesn't happen if I scroll with the touchpad. Yes please give it a try if you are able to replicate it.

**Comment by personalizedrefrigerator** [comment]
This diff *somewhat* helps:
```
diff --git a/packages/app-desktop/gui/NoteList/utils/useVisibleRange.ts b/packages/app-desktop/gui/NoteList/utils/useVisibleRange.ts
index 0042f49e1..af82b36bd 100644
--- a/packages/app-desktop/gui/NoteList/utils/useVisibleRange.ts
+++ b/packages/app-desktop/gui/NoteList/utils/useVisibleRange.ts
@@ -3,7 +3,7 @@ import { useMemo } from 'react';
 
 const useVisibleRange = (itemsPerLine: number, scrollTop: number, listSize: Size, itemSize: Size, noteCount: number) => {
 	const startLineIndexFloat = useMemo(() => {
-		return scrollTop / itemSize.height;
+		return Math.max(0, scrollTop / itemSize.height - 1);
 	}, [scrollTop, itemSize.height]);
 
 	const endLineIndexFloat = useMemo(() => {
```

It seems to prevent the main scrollable region from jumping, though doesn't prevent the **scrollbar** from jumping:

https://github.com/laurent22/joplin/assets/46334387/f5d275d7-64ef-4251-b45a-1e258d834d67

Logging the height of the top filler, with the patch applied, produces output similar to the following
```
fillerHeight 646
fillerHeight 680
fillerHeight 646
fillerHeight 612
fillerHeight 646
fillerHeight 612
fillerHeight 578
fillerHeight 612
fillerHeight 578
fillerHeight 544
```

Without the patch applied,
```
fillerHeight 544
fillerHeight 510
fillerHeight 544
fillerHeight 510
fillerHeight 544
fillerHeight 510
fillerHeight 544
fillerHeight 510
fillerHeight 544
fillerHeight 510
```

**Edit**: Setting the height to a fixed value doesn't seem to help.
**Edit**: The old `NoteList` works fine with the same version of Electron (tested by replacing NoteList2 with NoteList in `NoteListWrapper.tsx` and updating the import).

**Comment by personalizedrefrigerator** [comment]
Wrapping each `NoteListItem` in a `<div` with `style={{height: itemSize.height}}` seems to fix the issue! It seems that the list items temporarily have an incorrect size.

**Edit**: The fix in #9211 takes a similar approach, `NoteListItem`s outermost `<div>`s are styled to have `style={{ height: props.itemSize.height }}`.
